### DISCUSSION TITLE
Help with Robot Localization setup

### DISCUSSION THREAD

**Discussion by baradejoo** [FS-Online/Driverless-Competition-Simulator#347] [Q&A]
Hi, Im new in driverless car and I am going to setup my autonomous system code with simulator and im stucked at robot localization. Is here someone who can help with that config? I was trying a lot of option but still without any success. Only local odometry works with sensible covariance. Thanks in advance :)

**Comment by mateusz-lichota** [comment]
Hi Kamil!

I can try to help you with that. Are you using the robot_localization package? If so, what is your current config, and how does the trajectory look like in RVIZ?

**Comment by baradejoo** [comment]
Im using 3 nodes, navsat, 2x ekf (one for local, second one for global odometry). In navsat its basic official config (with magnetic_declination =0 and yaw_offset also =  0). In Local EKF im fusing /fsds/imu and /fsds/gss/fix (fix because in simulator is message type: geometry_message and ekf wants it to be nav_msgs), in global EKF there is fusion of /fsds/imu  and /odometry/gps (output from navsat). If you wanna I can give you access to my private repo (so far there is a mess) :D

**Reply by baradejoo** [reply]
In Rviz local odom looks good, but in  global position covariance increases with every moment of time  and without moving the car in simulator, the position seems to backwards

**Reply by baradejoo** [reply]
Its local odom: ![Screenshot from 2022-10-18 00-01-59](https://user-images.githubusercontent.com/49471097/196292253-52327ec5-87c4-47a1-aa47-7de0dca17cef.png)

**Reply by baradejoo** [reply]
gloabl odom: 
![Screenshot from 2022-10-18 00-04-03](https://user-images.githubusercontent.com/49471097/196292446-14f5d3c7-3edf-471d-b534-184e0fe6a026.png)

**Reply by baradejoo** [reply]
Additionally i see that /fsds/gps data are not available in rqt graph. But by rostopic echo i can see it.

**Comment by baradejoo** [comment]
Sorry for spam, but I've figured it out. The problem was the message name that robot localization (navsat) subscribes. Its a name: /gps/fix and  the simulator publishes /fsds/gps. In reference to documentation of robot localization changing the launcher (by adding <remap>) should works good but it doesn't. The solution is to add <remap> to launcher in bridge_ros or creating own node that translates names ;D
